### ISSUE TITLE
iperender crashes with bad alloc on prepared file

### ISSUE THREAD

**Issue by lorenzhs** [otfried/ipe#63]
File that triggers this behaviour:

``` xml
<?xml version="1.0"?>
<ipe version="70107" creator="Ipe 7.2.2">
<info created="D:20160419105135" modified="D:20160419105907"/>
<page>
<layer name="alpha"/>
<view layers="alpha" active="alpha"/>
<path layer="alpha" stroke="black">
16 832 m
132 8E6 l
32 832 l
32 832 l
h
</path>
</page>
</ipe>
```

Note the absurd `132 8E6` coordinate. Running `iperender -png foo.ipe foo.png`, the output is:

```
No text objects in document, no need to run Pdflatex.
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```

Backtrace:

``` cpp
(gdb) bt
#0  0x00007ffff6ec7478 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:55
#1  0x00007ffff6ec88fa in __GI_abort () at abort.c:89
#2  0x00007ffff77db32d in __gnu_cxx::__verbose_terminate_handler () at ../../../../src/libstdc++-v3/libsupc++/vterminate.cc:95
#3  0x00007ffff77d93b6 in __cxxabiv1::__terminate (handler=<optimized out>) at ../../../../src/libstdc++-v3/libsupc++/eh_terminate.cc:47
#4  0x00007ffff77d8469 in __cxa_call_terminate (ue_header=ue_header@entry=0x7ffff0000920) at ../../../../src/libstdc++-v3/libsupc++/eh_call.cc:54
#5  0x00007ffff77d8d55 in __cxxabiv1::__gxx_personality_v0 (version=<optimized out>, actions=<optimized out>, exception_class=5138137972254386944, ue_header=<optimized out>, context=0x7fffffffd340)
    at ../../../../src/libstdc++-v3/libsupc++/eh_personality.cc:676
#6  0x00007ffff7247ec3 in _Unwind_RaiseException_Phase2 (exc=exc@entry=0x7ffff0000920, context=context@entry=0x7fffffffd340) at ../../../src/libgcc/unwind.inc:62
#7  0x00007ffff724824b in _Unwind_RaiseException (exc=exc@entry=0x7ffff0000920) at ../../../src/libgcc/unwind.inc:131
#8  0x00007ffff77d960c in __cxxabiv1::__cxa_throw (obj=0x7ffff0000940, tinfo=0x7ffff7ab87b0 <typeinfo for std::bad_alloc>, dest=0x7ffff77d78e0 <std::bad_alloc::~bad_alloc()>)
    at ../../../../src/libstdc++-v3/libsupc++/eh_throw.cc:82
#9  0x00007ffff77d9b2c in operator new (sz=18446744073158195412) at ../../../../src/libstdc++-v3/libsupc++/new_op.cc:54
#10 0x00007ffff77d9b89 in operator new[] (sz=<optimized out>) at ../../../../src/libstdc++-v3/libsupc++/new_opv.cc:32
#11 0x00007ffff7f8f962 in ipe::Buffer::Buffer(int) () from /usr/local/lib/libipe.so.7.2.2
#12 0x00007ffff7ff28b2 in ipe::Thumbnail::saveRender(ipe::Thumbnail::TargetFormat, char const*, ipe::Page const*, int, double, bool, bool) () from /usr/local/lib/libipecairo.so.7.2.2
#13 0x00000000004010e1 in ?? ()
#14 0x00007ffff6eb4610 in __libc_start_main (main=0x400d40, argc=4, argv=0x7fffffffda08, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffd9f8) at libc-start.c:291
#15 0x0000000000401169 in ?? ()
```

The ipe gui copes fine. Version 7.2.2

**Comment by otfried** [comment]
How did you manage to create this file?  Ipe never writes something like "8E6" (even if the number is 8 Million, it would write 8000000), because PDF does not allow the E notation.

Something is seriously wrong here.  Perhaps some memory corruption, either in Ipe or on your computer?  

Also, the last two points on the path are the same - you are probably first clicking the last point with the left mouse key and then again with the right one.  You should only click it once (with the right mouse key).

**Comment by lorenzhs** [comment]
This file was created with a fuzzer (afl) which starts with a valid file and mucks about with it to see if it can make the program do something interesting (i.e. crash). After ipe crashed on me a few times in the last days I thought why not give that a whirl. I completely agree that it's a broken file, but it would be nice if this were caught instead of crashing. (It also found one other crash, if the XML file contains tags that have weird unicode symbols in them, ipe crashes)

I do tend to make the last-two-points mistake and tend to regret it soon after when editing the path - but in this case it was the result of the fuzzer monkeying around :)

**Comment by otfried** [comment]
Crash fixed in 7.2.3.
